src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,649 @@
+                    default:
+                        return resource[
+                            AzureResourceDisplayFields[displayField] // storage blob containers does not have url information in its resource properties, it is constructed based on the storage account blob url
+                        ];
+                }
+            } else {
+                return resource[AzureResourceDisplayFields[displayField]];
+            }
+        },
+        [displayField]
+    );
+
+    // after fetching the resources merge them with existing options in the dropdown
+    const mergeOptionsWithProps = useCallback(
+        (options, additionalOptions, selectedOption) => {
+            if (additionalOptions) {
+                // do the merging with existing options: add the additional options manually entered by user to the end if it is not in the fetched data
+                const existingOptionTexts =
+                    resourcesRef.current?.map((resource) =>
+                        getDisplayFieldValue(resource)
+                    ) || [];
+
+                const optionsToAdd = additionalOptions.filter(
+                    (additionalOption) =>
+                        existingOptionTexts.findIndex((existingOptionText) =>
+                            isValuesEqual(
+                                existingOptionText,
+                                additionalOption.text
+                            )
+                        ) === -1
+                );
+
+                if (optionsToAdd.length) {
+                    if (resourcesRef.current?.length > 0) {
+                        // add the freeform options header if the resources are fetched and exists
+                        options.push(freeformOptionsHeader);
+                    }
+                    options = options.concat(optionsToAdd);
+                }
+            }
+            if (selectedOption) {
+                const selectedOptionInNewOptions = options.find((option) =>
+                    isValuesEqual(option.text, selectedOption.text)
+                );
+                if (!selectedOptionInNewOptions) {
+                    const freeFromOptionsHeaderExist = options?.find(
+                        (option) =>
+                            option.itemType ===
+                                SelectableOptionMenuItemType.Header &&
+                            option.text === freeformOptionsHeaderText
+                    );
+                    if (!freeFromOptionsHeaderExist) {
+                        options.push(freeformOptionsHeader);
+                    }
+                    options = options.concat([selectedOption]);
+                } else {
+                    setSelectedKey(
+                        selectedOptionInNewOptions.data?.id ||
+                            selectedOption.text
+                    );
+                }
+            }
+            return options;
+        },
+        []
+    );
+
+    const optionsFromResources = () => {
```

Comment:
nit: please add a verb to the start of this name. `getOptionsFromResources`

Reply:
oh yeah, it was originally a useMemo variable forgot to rename it